dimitriosv/week4/app/components/Message.jsx

```diff
@@ -0,0 +1,46 @@
+var React = require('react');
+var ReactDOM = require('react-dom');
+var MessageActions = require('../actions/message-actions.js');
+
+
+var Message = React.createClass({
+    renderListOptions: function(messageId) {  return function(list,i) {
+        return (
+             <option key={list.listId+messageId} value={list.listId}>{list.listName}</option>
+        );
+    }
+    
+    },
+    render: function () {
+        return (<div>
+                    {this.props.messageIsArchived
```

Comment:
using inline if statements is nice, but in the `render` they make the code a bit harder to read. You can at least make sure to indent the markup correctly so it is clear where the if/else goes..